### PR TITLE
Implement failing cases.

### PR DESCRIPTION
**Pull request by isaachier**

### CODE REVIEW

File: util.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    int x = 0;
+    for (int i = 0; i < num_bits; ++i) {
+        x += 1;
+    }
+    assert(x > 0);
+    return static_cast<uint64_t>(1) << (64 - x);
+}
+
+int divide(int length)
```
Comment (ruslo) [inline util.cpp]: I think we don't need this (?)

**Review by ruslo — CHANGES_REQUESTED** [review]

File: util.cpp (hunk 1 above)
Comment (isaachier) [reply util.cpp]: True. That is an implementation of the function mentioned on the clang analyze website. It shouldn't make a difference, but another very similar bug.

Comment (ruslo) [reply util.cpp]: Let's focus on one bug, I think it will be quite enough. I can create another branch though.

File: util.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+#include <cassert>
+
+uint64_t shift(int num_bits)
+{
+    int x = 0;
+    for (int i = 0; i < num_bits; ++i) {
+        x += 1;
+    }
+    assert(x > 0);
```
Comment (ruslo) [inline util.cpp]: Move this block to original `foo`, this will be enough.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,6 @@
 cmake_minimum_required(VERSION 3.0)
-project(foo)
+project(analyze-bug)
 
-add_library(foo foo.cpp)
+add_library(util util.cpp)
```
Comment (ruslo) [inline CMakeLists.txt]: Having library is enough. To revert.

**Review by ruslo — CHANGES_REQUESTED** [review]

File: foo.h [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#ifndef FOO_H
```
Comment (ruslo) [inline foo.h]: We don't need this file

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,4 +1,3 @@
 cmake_minimum_required(VERSION 3.0)
-project(foo)
-
+project(analyze-bug)
```
Comment (ruslo) [inline CMakeLists.txt]: Not needed.

File: foo.cpp [hunk 6]
```diff
@@ -1,5 +1,13 @@
-#include <cstdint> // uint64_t
+#include "foo.h"
```
Comment (ruslo) [inline foo.cpp]: Not needed.

**Review by ruslo — CHANGES_REQUESTED** [review]

**Pull request merged**